docs/cli/dnx-migration.md

```diff
@@ -0,0 +1,152 @@
+replace them with CLI extensions. In this case, the `commands` node in `project.json` needs to be replaced by the 
+`tools` node and it needs to list the tools dependencies as explained in the 
+[CLI extensibility section](overview.md#extensibility). 
+
+After these things are done, you need to decide which type of portability you wish for you app. With .NET Core, we have 
+invested into providing a spectrum of portability options that you can choose from. For instance, you may want to have 
+a fully *portable* application or you may want to have a *self-contained* application. The former is more akin to the way 
+you've most likely seen .NET application to function: it needs a shared component to execute it on the target machine 
+(the .NET Core runtime). The latter doesn't require that, but you have to produce one application for each OS you wish to support. 
+These portability types and more are discussed in the [application portability type](app-types.md) document. 
+
+Once you make a call on what type of portability you want, you need to change your targeted framework(s). If you were 
+writing applications for .NET Core, you were most likely using `dnxcore50` as  your targeted framework. With the CLI 
+and the changes that the new [.NET Standard Library](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md) 
+brought, the framework needs to be one of the following:
+
+1. `netcoreapp1.0` - if you are writing applications on .NET Core (including ASP.NET Core applications)
+2. `netstandard1.5` - if you are writing class libraries for .NET Core
+
+Your `project.json` is now mostly ready. You need to go through your dependencies list and update the dependencies to 
+their newever versions, especially if you are using ASP.NET Core dependencies. If you used to use separate packages for BCL APIs
+you can use the runtime package as explained in the [application portability type](app-types.md) document. 
+
+Once you are ready, you can try restoring with `dotnet restore`. Depending on the the version of your dependencies, you 
+may encounter errors if NuGet cannot resolve the dependencies for one of the 
+targeted frameworks above. This is a "point-in-time" problem; as time progresses, more and more packages will include 
+support for these frameworks. For now, if you run into this, you can use the `imports` statement within the `framework` 
+node to specify to NuGet that it can restore the packages targeting the framework within the "imports" statement. In 
+general, specifying `dnxcore50` and `portable-net45+win8` in the `imports` statement should do the trick. The JSON snippet 
+below shows how this looks like:
+
+```json
+    "frameworks": {
+        "netstandardapp1.5": { 
+            "imports": ["dnxcore50", "portable-net45+win8"]
+        }
+    }
+```
+
+Running `dotnet build` will show any build errors, though there shouldn't be too much of them. After your code is 
```

Comment:
"Running `dotnet build` will show any build errors" - do you mean "may show build errors"?